streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -557,8 +558,8 @@ trait ZStream[-R, +E, +A] extends Serializable { self =>
    * Appends another stream to this stream. The concatenated stream will first emit the
    * elements of this stream, and then emit the elements of the `other` stream.
    */
-  final def concat[R1 <: R, E1 >: E, A1 >: A](other: ZStream[R1, E1, A1]): ZStream[R1, E1, A1] =
-    Stream(self, other).flatMap(identity)
+  final def concat[R1 <: R, E1 >: E, A1 >: A](other: => ZStream[R1, E1, A1]): ZStream[R1, E1, A1] =
+    ZStream(UIO.succeed(self), UIO(other)).flatMap(ZStream.unwrap)
```

Comment:
@jdegoes is there another combinator constellation we could use to make this nicer? e.g. a lazy constructor like `construct(head: A, rest: => List[A])`

Reply:
Awesome trick.